test-network-function/networking/suite.go

```diff
@@ -450,73 +450,66 @@ func listeningPortList(commandlisten []string, nodeOc *interactive.Context, list
 		if !ok {
-			tnf.ClaimFilePrintf(fmt.Sprintf("The port %d on protocol %s in pod %s is not declared.", k.port, k.protocol, listeningPorts[k]))
 			res[k] = listeningPorts[k]
 		}
 	}
 	return res
 }
 
 func testListenAndDeclared(env *config.TestEnvironment) {
-	declaredPorts := make(map[key]string)
-	listeningPorts := make(map[key]string)
-	undeclaredPorts := make(map[key]string)
-	var skippedPods []configsections.Pod
-	var failedPods []configsections.Pod
+	var failedcontainers []string
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestUndeclaredContainerPortsUsage)
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
-	OUTER:
-		for _, podUnderTest := range env.PodsUnderTest {
-			for i := 0; i < podUnderTest.ContainerCount; i++ {
-				err := declaredPortList(i, podUnderTest.Name, podUnderTest.Namespace, declaredPorts)
-				if err != nil {
-					tnf.ClaimFilePrintf("Failed to get declared port for container %d due to %v, skipping pod %s", i, err, podUnderTest.Name)
-					skippedPods = append(skippedPods, *podUnderTest)
-					continue OUTER
-				}
+		for _, cut := range env.ContainersUnderTest {
+			declaredPorts := make(map[key]bool)
+			listeningPorts := make(map[key]bool)
+			container, _ := strconv.Atoi(cut.ContainerUID)
+			err := declaredPortList(container, cut.PodName, cut.Namespace, declaredPorts)
+			if err != nil {
+				tnf.ClaimFilePrintf("Failed to get declared port for container %d due to %v, skipping pod %s", cut.ContainerUID, err, cut.PodName)
+				continue
 			}
-
-			nodeName := podnodename.NewPodNodeName(common.DefaultTimeout, podUnderTest.Name, podUnderTest.Namespace)
+			nodeName := podnodename.NewPodNodeName(common.DefaultTimeout, cut.PodName, cut.Namespace)
 			context := env.GetLocalShellContext()
 			test, err := tnf.NewTest(context.GetExpecter(), nodeName, []reel.Handler{nodeName}, context.GetErrorChannel())
 			gomega.Expect(err).To(gomega.BeNil())
 			test.RunAndValidate()
 			nodeOc := env.NodesUnderTest[nodeName.GetNodeName()].DebugContainer.GetOc()
-			x := podUnderTest.ContainerList[0]
-			containerPID := utils.GetContainerPID(nodeName.GetNodeName(), nodeOc, x.ContainerUID, x.ContainerRuntime)
+			containerPID := utils.GetContainerPID(nodeName.GetNodeName(), nodeOc, cut.ContainerUID, cut.ContainerRuntime)
 
 			commandlisten := []string{utils.AddNsenterPrefix(containerPID), commandportlisten}
 
 			err = listeningPortList(commandlisten, nodeOc.Context, listeningPorts)
 			if err != nil {
-				tnf.ClaimFilePrintf("Failed to get listening port for pod name %s in pod namespace %s due to %v, skipping this pod", podUnderTest.Name, podUnderTest.Namespace, err)
+				tnf.ClaimFilePrintf("Failed to get listening port for pod name %s in pod namespace %s on container %s due to %v, skipping this pod", cut.PodName, cut.Namespace, cut.ContainerUID, err)
+				failedcontainers = append(failedcontainers, fmt.Sprintf("podName: %s, Namespace: %s, container:%s", cut.PodName, cut.Namespace, cut.ContainerUID))
```

Comment:
Please, do not use cut.ContainerUID in these error traces, as it's just a number without a meaning for the customer. Change this trace to:
```
tnf.ClaimFilePrintf("Failed to get listening ports for container %s pod %s ns %s due to err %s", cut.ContainerName, cut.PodName, cut.Namespace, err)
```